cdap-docs/developers-manual/source/advanced/adapters.rst

```diff
@@ -0,0 +1,140 @@
+=========================
+The stream-conversion Adapter type regularly reads data from a source Stream and writes
+it to a TimePartitionedFileSet as avro files. You may want to use this Adapter if you
+want to run partitioned Hive or Impala queries on data in a Stream, or if you have set a
+TTL on a Stream for real-time processing, but want to keep all Stream data around in
+another format for archiving purposes. Each Stream event in translated as a line in
+the output fileset. Stream format and schema must be given as adapter properties.
+The output schema will add a required ``ts`` field of type long for the timestamp of the
+event, as well as nullable string fields for any headers given in the adapter properties. 
+
+Adapters of the stream-conversion type require the following adapter properties:
+
+  * frequency: how often the Adapter should run, as a number followed by a number. For
+    example, 5m means the Adapter will run every five minutes, and 2h means the Adapter
+    will run every two hours. Note that this must be a number that divides evenly into
+    the time unit given. For example, 30m is valid, but 13m is not because an hour does
+    not evenly divide into 13 minute chunks.
+  * source.name: the name of the Stream to read from.
+  * source.format.name: the name of the format of the Stream body.
+  * source.format.settings: settings of the format as a JSON Object of string to string.
+  * source.schema: the schema of the Stream body.
+  * sink.name: the name of the TimePartitionedFileSet to write to. 
+
+The following adapter properties are optional:
+
+  * mapper.resources.memory.mb: how much memory each mapper of the mapreduce job should use.
+  * mapper.resources.vcores: how many virtual cores each mapper of the mapreduce job should use.
+  * headers: which stream headers should be included in the output. 
+
+The following sink properties are required:
+
+  * explore.table.property.avro.schema.literal: the output schema with timestamp and headers included. 
+
+Each time the Adapter runs, it will read stream events from a time range, write to the fileset, and add those
+files as a partition. The time range is calculated based on the frequency of the Adapter and the logical
+start time of the Adapter run. For example, if the Adapter is configured to run every hour, each run of the
+Adapter will read events starting from the logical start time minus an hour and ending at the logical start time.
+The logical start time is the time at which the schedule is triggered. For an hour frequency, this means
+the logical start time will be on the hour, every hour. Even if the run actually begins at 5:03, the logical
+start will be 5:00 and the run will read Stream events that were added from 4:00 to 4:59.
+
+Because there are so many properties, and because properties like schema must be string values of
+escaped JSON Objects, it is highly recommended that you use the CLI to create stream-conversion adapters.
+For example::
```

Comment:
As there are so many properties, and because properties such as schema must be string values of
escaped JSON Objects, it is highly recommended that you use the CLI to create stream-conversion Adapters.